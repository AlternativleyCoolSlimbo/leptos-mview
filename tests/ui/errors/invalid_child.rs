use leptos_mview::mview;

fn main() {
    let a = "a";
    mview! {
        (a)
    };
}

// checking that it doesn't suggest adding `|_| {value}`
fn not_impl_intoview() {
    // &&str doesn't impl IntoView, &str does
    // should be `{*value}`
    let value: &&str = &"hi";
    _ = mview! {
        span (
            {value}
        )
    };

    // forgot to call `.collect_view()`
    let values: Vec<&'static str> = vec!["hi", "bye", "howdy", "hello", "hey"];
    _ = mview! {
        ul {
            {values
                .into_iter()
                .map(|val: &str| {
                    mview! { li({val}) }
                })
            }
        }
    }
}

fn extra_semicolons() {
    _ = mview! {
        div { "hi there" };
        span;
    };
}

fn unreachable_code() {
    _ = mview! {
        div {
            {todo!()}
        }
    }
}
