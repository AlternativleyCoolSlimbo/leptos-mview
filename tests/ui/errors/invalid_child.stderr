error: invalid child: expected literal, block, bracket or element
 --> tests/ui/errors/invalid_child.rs:6:9
  |
6 |         (a)
  |         ^

error: extra semi-colon found
  --> tests/ui/errors/invalid_child.rs:37:27
   |
37 |         div { "hi there" };
   |                           ^
   |
   = help: remove this semi-colon

error[E0277]: expected a `Fn()` closure, found `str`
  --> tests/ui/errors/invalid_child.rs:17:14
   |
15 |       _ = mview! {
   |  _________-
16 | |         span (
17 | |             {value}
   | |              ^^^^^ expected an `Fn()` closure, found `str`
18 | |         )
19 | |     };
   | |_____- required by a bound introduced by this call
   |
   = help: the trait `std::ops::Fn<()>` is not implemented for `str`
   = note: wrap the `str` in a closure with no arguments: `|| { /* code */ }`
   = help: the trait `IntoView` is implemented for `&'static str`
   = note: required for `&str` to implement `FnOnce<()>`
   = note: required for `&&str` to implement `IntoView`
note: required by a bound in `leptos::HtmlElement::<El>::child`
  --> $CARGO/leptos_dom-0.6.3/src/html.rs
   |
   |     pub fn child(self, child: impl IntoView) -> Self {
   |                                    ^^^^^^^^ required by this bound in `HtmlElement::<El>::child`

error[E0277]: expected a `Fn()` closure, found `std::iter::Map<std::vec::IntoIter<&str>, {closure@$DIR/tests/ui/errors/invalid_child.rs:27:22: 27:33}>`
  --> tests/ui/errors/invalid_child.rs:25:14
   |
23 |        _ = mview! {
   |  __________-
24 | |          ul {
25 | |              {values
   | | ______________^
26 | ||                 .into_iter()
27 | ||                 .map(|val: &str| {
28 | ||                     mview! { li({val}) }
29 | ||                 })
   | ||__________________^ expected an `Fn()` closure, found `std::iter::Map<std::vec::IntoIter<&str>, {closure@$DIR/tests/ui/errors/invalid_child.rs:27:22: 27:33}>`
30 | |              }
31 | |          }
32 | |      }
   | |______- required by a bound introduced by this call
   |
   = help: the trait `std::ops::Fn<()>` is not implemented for `std::iter::Map<std::vec::IntoIter<&str>, {closure@$DIR/tests/ui/errors/invalid_child.rs:27:22: 27:33}>`
   = note: wrap the `std::iter::Map<std::vec::IntoIter<&str>, {closure@$DIR/tests/ui/errors/invalid_child.rs:27:22: 27:33}>` in a closure with no arguments: `|| { /* code */ }`
   = help: the following other types implement trait `IntoView`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and $N others
   = note: required for `std::iter::Map<std::vec::IntoIter<&str>, {closure@$DIR/tests/ui/errors/invalid_child.rs:27:22: 27:33}>` to implement `IntoView`
note: required by a bound in `leptos::HtmlElement::<El>::child`
  --> $CARGO/leptos_dom-0.6.3/src/html.rs
   |
   |     pub fn child(self, child: impl IntoView) -> Self {
   |                                    ^^^^^^^^ required by this bound in `HtmlElement::<El>::child`
